### DISCUSSION TITLE
Features and improvement ideas

### DISCUSSION THREAD

**Discussion by fryktoria** [fryktoria/On-the-Fly-Shortest-Path#8] [Announcements]
<hr /><em>This discussion area refers to comments on the functionality of existing features as well as ideas on how to implement new functionality for the plugin.</em>

**Comment by parydeogracias** [comment]
Hi, congratulations on your achievement, attenuation calculation not very common in qgis. 

As it stands, it's not bad for point-to-point work, but it lacks the ability to take couplers into account when calculating attenuation on GPON. Also, in the gpon configuration, splice boxes are not at fixed distances.

Your plugin will reach a milestone by integrating the Gpon part.
Many thanks!!!!!

**Reply by fryktoria** [reply]
Thank you for your kind words. Regarding GPON, there are several implementation models used by operators around the world, such as using redundant paths with 2:x splitters, one or two or three level splitting etc. There are also different use cases, such as purely urban versus rural, where in the urban case the loss of the fiber is minimal compared to the loss of the splitters. All these complexities could be covered only by a specialized GPON design software, rather than a general-purpose GIS such as QGIS. 

Considering that a GPON designer implements a specific model, the total loss of the splitters would be more-or-less fixed. This value can be set in the "Fixed loss" parameter and provide the user with an indication of the contribution of splitters and other loss elements in the total attenuation of the path. 

This "Fixed loss" parameter was as much as I could think of, as a simple and generic solution. Yet, if you have an idea about how to enhance the tool with a GPON functionality, I would be happy to hear from you.

**Reply by parydeogracias** [reply]
Thank you for your reply, in the Word file I tried to add points while remaining brief, which could be done in a PON view. Moreover, by not going to the customer level (ONT), the only wavelength to be used will be 1490nm and the linear attenuation that will be declared will only be that corresponding to this length.
[GPON_XG(S)-PON_Splitters.docx](https://github.com/user-attachments/files/15927140/GPON_XG.S.-PON_Splitters.docx)


I've also included a screenshot of the file used for this calculation for your convenience.
![Capture d'écran 2024-06-21 115357](https://github.com/fryktoria/On-the-Fly-Shortest-Path/assets/26378403/97d3d487-f7d7-4af4-be02-c8239c08c2d0)

Thank you very much!

**Reply by fryktoria** [reply]
To make sure that I have understood your proposal, let me compare with the current functionality:

1. There is a loss component coming from the start and end connectors. This loss does not depend on the length of the shortest path.
2. There is a loss component that depends on the length, coming from the fiber attenuation and the regular splices
3. There is an additional fixed loss component, defined by the user. This may encompass the loss of splitters or any other component and does not depend on the length of the shortest path. This Fixed Loss can be added at will to the sum of the previous two loss components, by clicking on the "Add Fixed loss" checkbox of the Panel.

According to your proposal, the fixed loss component is further broken down in losses of the following components:

1. Splitter Level 1
2. Splitter Level 2
3. Splitter Level 3
4. Loss of connectors at Splitter Level 1 by setting the number of connectors, to be multiplied by the "Connector loss" parameter of the original functionality
5. Loss of connectors at Splitter Level 2 as above
6. Loss of connectors at Splitter Level 3 as above
7. Loss of splice boxes associated with the GPON, by setting the number of connectors, to be multiplied by the "Splice loss" parameter of the original functionality
8. On top of the above GPON elements, we will also have the original fixed loss component.

The sum of all the above, i.e the GPON elements plus the fixed loss will constitute a new "Total fixed loss", which will be the value that will be added to the result when the checkbox "Add Fixed loss" is checked. This amount does not depend on the length of the shortest path.

Have I understood correctly?

**Reply by parydeogracias** [reply]
That's right! Thank you so much for taking the time to read me!!!!!!!!

**Reply by fryktoria** [reply]
Well, it is not too much work to add a few parameters. I am not negative but my main concern is that all these values add up to a single fixed number, which does not depend to the calculated length and therefore could be calculated with an external application, such as Excel in a much more efficient way. I would consider the functionality integrated into the plugin similar to a low-feature calculator. 

Not only that, but if a user has filled in all these values and for some reason something changes, e.g. using a different wavelength or calculating the attenuation at the reverse direction from ONT to OLT, all parameters will need to be modified and eventually, in spite of the presence of all those parameters,  the user will prefer to use the cumulative "other fixed losses", which is already covered by the existing model. 

If we were thinking about covering multiple scenarios, this would mean that we would have to create sets of parameters (e.g. "profiles") where the user would be able to select one of those profiles at a time. But this functionality goes far beyond the notion of configuration parameters.

Could you please provide some examples of real-life use cases where these parameters would help in the daily operations and justify the additional development effort?

**Reply by fryktoria** [reply]
It seems that it was relatively easy to implement. I prefered to use "profiles" to accomodate various cases such as 1490nm, 1550nm, backbone only etc.  The profiles operate not only on the GPON elements but to all fiber loss components.  I just need to test that it works both on Linux and Windows before I release a new version.
![image](https://github.com/fryktoria/On-the-Fly-Shortest-Path/assets/58485561/b427b7b6-ea4a-4e4b-a0fc-12e5a0b13769)

**Reply by parydeogracias** [reply]
That's great news if you need a tester. I'm at your disposal. Thank you!

**Reply by fryktoria** [reply]
What are the values that you are using for 1550nm or other wavelengths?

**Comment by parydeogracias** [comment]
# GPON Parameters

## Loss Metrics
- **Connectors:** 0.3 dB
- **Loss/km:** 0.25 dB
- **Splice Loss:** 0.2 dB

## Splitter Loss
- **1x2:** 3.6 dB
- **1x4:** 6.9 dB
- **1x8:** 10.2 dB
- **1x16:** 13.5 dB
- **1x32:** 16.5 dB
- **1x64:** 20 dB

**Comment by parydeogracias** [comment]
![Capture d'écran 2024-07-03 124815](https://github.com/fryktoria/On-the-Fly-Shortest-Path/assets/26378403/ea786656-8bb4-4873-96e7-2e0da1b6551e)

Is it possible to take into account the fact that degree projections are used? As you can see from the screenshot, the attenuation does not conform.

**Reply by fryktoria** [reply]
The distance calculations use the ellipsoid which is indirectly selected using the CRS. All internal QGIS calculations are performed in meters and are transformed to the units selected in the plugin "Network Analysis" tab of the configuration window. Degrees are not used.  Can you provide the parameters that produced the above loss values? Also, if you consider that there is a bug, please report it in the Github Issues. Thank you.

**Reply by parydeogracias** [reply]
I'll report it in the Github Issues.
![image](https://github.com/fryktoria/On-the-Fly-Shortest-Path/assets/26378403/584dbe30-813c-47ee-810b-2a6103975139)
